### DISCUSSION TITLE
freshness slo

### DISCUSSION THREAD

**Discussion by freeseacher** [fluxcd/flux2#2069] [General]
Any ideas how to implement freshness SLO for flux?
As user i want to know how much should i actually wait before my git configuration was really applied. 
As SRE i want to know how percentage of time configurations was not in sync after requested interval by user. 

freshness slo is defined here https://sre.google/workbook/implementing-slos/ 

The proportion of the data that was updated more recently than some time threshold. Ideally this metric counts how many times a user accessed the data, so that it most accurately reflects the user experience.

**Comment by freeseacher** [comment]
I'm thinking of sending to pushgeteway metric with repo name during ci phase of deployment, than get information about actual deployment time, it is now can be retrived from flux itself.

**Comment by stefanprodan** [comment]
The reconciliation duration of each resource (GitRepository, Kustomization, HelmRelease, etc) can be tracked with Prometheus. Check out: https://fluxcd.io/docs/guides/monitoring/#metrics

**Reply by freeseacher** [reply]
that is just part of the story. User expectations are defined as diff time between merge to master and their change on prod. there so many things that can go wrong to check only flux metrics. 
The story of this discussion has begun when one repo was renamed and changed its url. Users has updated one repo but flux was watching to another

**Reply by kingdonb** [reply]
I am elaborating on this idea here: https://github.com/fluxcd/flux2/discussions/2238#discussioncomment-1848228

I think you're right, the freshness is the metric to measure if your team delivers Flux as a Service 👍 and coverage
